recap/converters/json_schema.py

```diff
@@ -65,12 +69,22 @@ def _parse(
             self.json_registry = resource @ self.json_registry
             extra_attrs["alias"] = alias_strategy(resource_id)
 
+        # Special handling for "type" defined as a list of strings like
+        # {"type": ["string", "boolean"]}
+        if "type" in json_schema and isinstance(json_schema["type"], list):
+            types = [self._parse(s, alias_strategy) for s in json_schema["type"]]
+            return UnionType(types, **extra_attrs)
+
         match json_schema:
             case {"type": "object", "properties": properties}:
                 fields = []
                 for name, prop in properties.items():
                     field = self._parse(prop, alias_strategy)
-                    if name not in json_schema.get("required", []):
+                    # Make type nullable if it's not required, but avoid double nullable types
+                    if (
+                        name not in json_schema.get("required", [])
+                        and not field.is_nullable()
```

Comment:
I think there's a nuance that's not captured here. `is_nullable` checks `UnionType` with a `NullType` in it. It's not checking if there's a default value. Without a default value, the field is still required (a value *must* be set, even if it's null).

A JSON schema field with `{"type": ["string", "null"]}` that's not listed in `required` should be optional (i.e. it should have a default null, even if it's not set explicitly in the JSON schema). I don't think current logic accounts for this.

Reply:
Ah good catch, you are correct. Update the logic here, and the relevant test